truffleruby/src/main/java/org/truffleruby/core/binding/BindingLayout.java

```diff
@@ -13,6 +13,8 @@
 import com.oracle.truffle.api.object.DynamicObject;
 import com.oracle.truffle.api.object.DynamicObjectFactory;
 import com.oracle.truffle.api.object.dsl.Layout;
+import com.oracle.truffle.api.object.dsl.Nullable;
+
 import org.truffleruby.core.basicobject.BasicObjectLayout;
```

Comment:
It would be great to have 1-2 sentences here explaining a bit how Binding are represented.
Something like:
```java
/**
 * Bindings capture the frame from where they are called, which is stored in +frame+,
 * but also have their own frame(s) in +extras+ to store variables added in the Binding,
 * which must not be added in the captured frame (MRI semantics).
 * We choose to create a new extra frame for each new variable to respect the semantics
 * of Binding#dup where the copy shows updates of existing variables but not new variables
 * added after the copy.
 */
```